server/app/controllers/admin/AdminApplicationController.java

```diff
@@ -265,12 +304,25 @@ public Result show(Http.Request request, long programId, long applicationId)
   /** Return a paginated HTML page displaying (part of) all applications to the program. */
   @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
   public Result index(
-      Http.Request request, long programId, Optional<String> search, Optional<Integer> page)
+      Http.Request request,
+      long programId,
+      Optional<String> search,
+      Optional<Integer> page,
+      Optional<String> fromDate,
+      Optional<String> untilDate)
       throws ProgramNotFoundException {
     if (page.isEmpty()) {
-      return redirect(routes.AdminApplicationController.index(programId, search, Optional.of(1)));
+      return redirect(
+          routes.AdminApplicationController.index(
+              programId, search, Optional.of(1), fromDate, untilDate));
     }
 
+    TimeFilter submitTimeFilter =
+        TimeFilter.builder()
+            .setFromTime(parseDateFromQuery(dateConverter, fromDate))
```

Comment:
I'd be careful here with time zones. Is it doing midnight local (browser), midnight server time, UTC, or the current time.  Since we're not asking for a timestamp or timezone, it's unclear.